src/activation.rs

```diff
@@ -89,19 +89,33 @@ pub fn receive_descriptors(unset_env: bool) -> Result<Vec<FileDescriptor>, SdErr
         env::remove_var("LISTEN_FDNAMES");
     }
 
+    // Parse `$LISTEN_PID` if present.
+    if let Err(env::VarError::NotPresent) = pid {
+        return Ok(vec![]);
+    }
     let pid = pid
         .context("failed to get LISTEN_PID")?
         .parse::<u32>()
         .context("failed to parse LISTEN_PID")?;
+    let current_pid = process::id();
+    if pid != current_pid {
+        log::warn!(
```

Comment:
I think encountering this is not really broken, so warn seems too high severity to me. I'd use some lower severity with a message along the lines of "systemd activation environment is set but not meant for this process".

Basically, I'd treat it as pretty normal case but logging it still has value in case someone is debugging systemd itself.

Reply:
Yes, this is mostly meant for debugging.
I've lowered the log level and reworded the message a bit.
Thanks for the feedback.